SwiftCompilerSources/Sources/Optimizer/FunctionPasses/EscapeInfoDumper.swift

```diff
@@ -12,38 +12,45 @@
 
 import SIL
 
+struct EIWVisitor : EscapeInfoWalkerVisitor {
+  var results: Set<String> =  Set()
+  
+  mutating func visitUse(operand: Operand, path: Path, state: State) -> UseVisitResult {
+    if operand.instruction is ReturnInst {
+      results.insert("return[\(path)]")
+      return .ignore
+    }
+    return .continueWalk
+  }
+  
+  mutating func visitDef(def: Value, path: Path, state: State) -> DefVisitResult {
+    guard let arg = def as? FunctionArgument else {
+      return .continueWalkUp
+    }
+    results.insert("arg\(arg.index)[\(path)]")
+    return .walkDown
+  }
+}
+
 /// Dumps the results of escape analysis.
 ///
 /// Dumps the EscapeInfo query results for all `alloc_stack` instructions in a function.
 ///
 /// This pass is used for testing EscapeInfo.
 let escapeInfoDumper = FunctionPass(name: "dump-escape-info", {
   (function: Function, context: PassContext) in
-
+  
   print("Escape information for \(function.name):")
-
-  var escapeInfo = EscapeInfo(calleeAnalysis: context.calleeAnalysis)
-
+  
+  var walker = EscapeInfoWalker(calleeAnalysis: context.calleeAnalysis, visitor: EIWVisitor())
```

Comment:
You can move the walker into the inner loop. Performance is not critical here at all.